tests/runtests.pl

```diff
@@ -2737,12 +2737,42 @@ sub responsive_httptls_server {
     return &responsiveserver($proto, $ipvnum, $idnum, $ip, $port);
 }
 
+#######################################################################
+# Kill the processes that still lock files in a directory
+#
+sub clearlocks {
+    my $dir = $_[0];
+    my $done = 0;
+
+    if(pathhelp::os_is_win()) {
+        $dir = pathhelp::sys_native_abs_path($dir);
+        $dir =~ s/\//\\\\/g;
+        my $handle = "handle.exe";
+        if($ENV{"PROCESSOR_ARCHITECTURE"} =~ /64$/) {
+            $handle = "handle64.exe";
+        }
+        my @handles = `$handle $dir -accepteula -nobanner`;
+        for $handle (@handles) {
+            if($handle =~ /^(\S+)\s+pid:\s+(\d+)\s+type:\s+(\w+)\s+([0-9A-F]+):\s+(.+)\r\r/) {
```

Comment:
why check for two carriage returns at the end, is this some weird quirk of msys i forgot about

Reply:
That is specific to the output format of the handle tool.

Reply:
I found something possibly related in some of my old code, if it's any help. It runs from msys perl to get service paths and grep to remove the \r\r and trailing whitespace and blank lines and header in wmic output
~~~perl
my @commands = grep { s/[ \t\r\n]*$//mg; !/^(PathName|)$/ }
               `wmic service GET PathName`;
~~~

Reply:
Thanks, I will remember this for future issues. I guess in this case the existing regex is just fine for the specific line parsed.